planner/core/rule_join_elimination.go

```diff
@@ -73,33 +73,43 @@ func (o *outerJoinEliminator) extractInnerJoinKeys(join *LogicalJoin, innerChild
 	return expression.NewSchema(joinKeys...)
 }
 
-func (o *outerJoinEliminator) isAggColsAllFromOuterTable(outerPlan LogicalPlan, aggCols []*expression.Column) (bool, error) {
+func (o *outerJoinEliminator) isAggColsAllFromOuterTable(outerPlan LogicalPlan, aggCols []*expression.Column) bool {
 	if len(aggCols) == 0 {
-		return false, nil
+		return false
 	}
-	for _, col := range aggCols {
-		columnName := &ast.ColumnName{Schema: col.DBName, Table: col.TblName, Name: col.ColName}
-		c, err := outerPlan.Schema().FindColumn(columnName)
-		if err != nil || c == nil {
-			return false, err
+	for _, aggCol := range aggCols {
+		isAggColInOuterSchema := false
+		for _, outerCol := range outerPlan.Schema().Columns {
+			if aggCol.UniqueID == outerCol.UniqueID {
+				isAggColInOuterSchema = true
+				break
+			}
+		}
+		if !isAggColInOuterSchema {
+			return false
 		}
 	}
-	return true, nil
+	return true
 }
 
 // check whether schema cols of join's parent plan are all from outer join table
-func (o *outerJoinEliminator) isParentColsAllFromOuterTable(outerPlan LogicalPlan, parentSchema *expression.Schema) (bool, error) {
-	if parentSchema == nil {
-		return false, nil
-	}
-	for _, col := range parentSchema.Columns {
-		columnName := &ast.ColumnName{Schema: col.DBName, Table: col.TblName, Name: col.ColName}
-		c, err := outerPlan.Schema().FindColumn(columnName)
-		if err != nil || c == nil {
-			return false, err
+func (o *outerJoinEliminator) isParentColsAllFromOuterTable(outerPlan LogicalPlan, parentCols []*expression.Column) bool {
```

Comment:
Can we merge this function with `isAggColsAllFromOuterTable`?

Reply:
No, we can't. `isParentColsAllFromOuterTable` is the base condition for all two     eliminating rules. But `isAggColsAllFromOuterTable ` is for only one.

Reply:
Oh, I mean these two functions look the same except there parameter name.

Reply:
You are right..